lib/CryptoPluginApi/extensionplugins.cpp

```diff
@@ -15,10 +15,642 @@
 
 using namespace Sailfish::Crypto;
 
+/*!
+  \class CryptoPlugin
+  \brief Specifies an interface which provides a variety of cryptographic
+         operations to clients which will be exposed via the Sailfish OS
+         Crypto API.
+
+  The CryptoPlugin type specifies an interface which provides encryption,
+  decryption, signing, verification, key derivation, key pair generation,
+  random data generation, and digest calculation.
+
+  The interface also includes methods related to key storage and retrieval,
+  for those plugins which include external key storage capability (which
+  offers the maximum security for stored keys as they need never be pulled
+  into the secrets daemon process address space).  Note that if a plugin
+  implements these methods, it must also implement the
+  Sailfish::Secrets::EncryptedStoragePlugin interface from the Sailfish OS
+  Secrets Extension Plugin API.  Any plugin which implements both of these
+  interfaces is known as a Crypto Storage Plugin (for an example of such a
+  plugin, please see the ExampleUsbTokenPlugin in the source tree).
+
+  Plugin implementers must be aware that the information reporting methods
+  (encryptionType(), and canStoreKeys()) will be invoked from the main
+  thread of the secrets daemon, while the various interface operation methods
+  will be invoked from a separate thread.  Plugins are loaded and plugin
+  instances are constructed in the main thread.
+
+  In order to implement a Crypto extension plugin, plugin implementers should
+  specify the following in their .pro file:
+  \code
+  CONFIG += link_pkgconfig
+  PKGCONFIG += sailfishcryptopluginapi
+  \endcode
+
+  The CryptoPlugin class extends the Sailfish::Secrets::PluginBase abstract
+  base class, and the sailfishcryptopluginapi pkgconfig file will entail
+  a dependency upon sailfishsecretspluginapi and ailfishsecrets, as
+  well as sailfishcrypto.
+ */
+
+/*!
+ * \enum CryptoPlugin::EncryptionType
+ *
+ * This enum defines the types of encryption capability which may be offered by plugins
+ *
+ * \value NoEncryption No encryption is performed
+ * \value SoftwareEncryption Encryption is performed by "normal" rich execution environment application
+ * \value TrustedExecutionSoftwareEncryption Encryption is performed by trusted execution environment application
```

Comment:
Maybe defining « trusted » here, I have no clue what is a normal rich and a trusted environment.

Reply:
TEE is an industry term for a secure area inside the main processor.  I've added the acronyms explicity now to make it more clear.